lib/Transforms/MemoryBanking.cpp

```diff
@@ -409,6 +409,151 @@ struct BankAffineStorePattern
+        return failure();
+      }
+    } else {
+      auto &dstBanks = memoryToBanks.find(dstMem)->second;
+      if (failed(copyFromUnbankedToBanks(rewriter, loc, srcMem, dstBanks,
+                                         bankingFactor, bankingDimension)))
+        return failure();
+    }
+
+    rewriter.eraseOp(copyOp);
+
+    return success();
+  }
+
+private:
+  uint64_t bankingFactor;
+  unsigned bankingDimension;
+  DenseMap<Value, SmallVector<Value>> &memoryToBanks;
+  /// Helper function to create a subview with round-robin partitioning.
+  /// Generates offsets, sizes, and strides for a specific bank and creates the
+  /// subview.
+  Value createRoundRobinSubview(PatternRewriter &rewriter, Location loc,
+                                Value mem, ArrayRef<int64_t> shape,
+                                size_t bankIndex, uint64_t bankingFactor,
+                                unsigned bankingDimension) const {
+    int64_t rank = shape.size();
+    // Initialize SubView offsets, sizes, strides for each dimension with
+    // default values.
+    SmallVector<OpFoldResult> offsets(rank, rewriter.getIndexAttr(0));
+    SmallVector<OpFoldResult> sizes(rank);
+    SmallVector<OpFoldResult> strides(rank, rewriter.getIndexAttr(1));
+
+    for (int64_t d = 0; d < rank; ++d) {
+      int64_t dimSize = shape[d];
+      if (d == bankingDimension) {
+        // Round-robin partition: offset = bankIndex, size = chunkSize, stride =
+        // bankingFactor
+        int64_t chunkSize = dimSize / (int64_t)bankingFactor;
+        offsets[d] = rewriter.getIndexAttr(bankIndex);
+        sizes[d] = rewriter.getIndexAttr(chunkSize);
+        strides[d] = rewriter.getIndexAttr(bankingFactor);
+      } else {
+        // Non-banked dimension: copy the entire range, unit stride
+        offsets[d] = rewriter.getIndexAttr(0);
+        sizes[d] = rewriter.getIndexAttr(dimSize);
+        strides[d] = rewriter.getIndexAttr(1);
+      }
+    }
+
+    return rewriter.create<memref::SubViewOp>(loc, mem, offsets, sizes,
+                                              strides);
+  }
+
+  /// Copy from source memory reference to destination memory reference when the
+  /// source memory is banked into `srcBanks` and `dstMem` is unbanked. All
+  /// `srcBanks` have the same MemRefType; the unbanked `dstMem` has the same
+  /// shape as the source's "original" shape.
+  LogicalResult copyFromBanksToUnbanked(PatternRewriter &rewriter, Location loc,
+                                        ArrayRef<Value> srcBanks, Value dstMem,
+                                        uint64_t bankingFactor,
+                                        unsigned bankingDimension) const {
+    MemRefType dstType = dyn_cast<MemRefType>(dstMem.getType());
+    if (!dstType || dstType.getRank() <= bankingDimension)
+      return failure();
+
+    ArrayRef<int64_t> dstShape = dstType.getShape();
+    int64_t chunkSize = dstShape[bankingDimension] / (int64_t)bankingFactor;
```

Comment:
Please avoid implicit casts (here and above and below).

```suggestion
    int64_t chunkSize = dstShape[bankingDimension] / static_cast<int64_t>(bankingFactor);
```